### PR TITLE
[LIVE-11886][LLM] Custom lock screen preview crashing with some pictures

### PR DESCRIPTION
**Pull request by ofreyssinet-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - LLM custom lock screen preview screen

### 📝 Description

There were some errors in the implementation of the picture rescaling, before the cropping, which led to bad cropping parameters, leading to an (uncaught) native crash of `expo-image-manipulator`.
The implementation is fixed and unit tests have been added.

Here's the fixed result with the picture that was crashing in the reported issue:
Simulator Screenshot - iPhone SE (3rd generation) - 2024-03-22 at 18 41 46

### ❓ Context

- **JIRA or GitHub link**: [LIVE-11886] 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-11886]: https://ledgerhq.atlassian.net/browse/LIVE-11886?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getRescaledDimensions.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  return {
+    width:
+      limitingDimensions === "width"
+        ? containerDimensions.width
+        : Math.floor(imageDimensions.width * resizeRatio),
+    height:
+      limitingDimensions === "height"
+        ? containerDimensions.height
+        : Math.floor(imageDimensions.height * resizeRatio),
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getRescaledDimensions.ts:46]: This is the part of the implementation that was fixed, previously we were always returning the `Math.floor()` which was leading in some cases to a rounding error, and returned dimensions that are smaller than expected.

File: apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getCenteredCropParams.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export function getCenteredCropParams(
+  imageDimensions: ImageDimensions,
+  cropDimensions: ImageDimensions,
+): { originX: number; originY: number; width: number; height: number } {
+  if (
+    cropDimensions.width > imageDimensions.width ||
+    cropDimensions.height > imageDimensions.height
+  ) {
+    throw new ImageCropError("The cropping dimensions must be smaller than the image dimensions");
+  }
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getCenteredCropParams.ts:20]: This is the part that was added in this PR

File: apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getCenteredCropParams.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * Returns the parameters to crop an image in the center.
+ * @param imageDimensions The dimensions of the image to crop
+ * @param cropDimensions The dimensions of the container to crop in
+ * @returns The parameters to crop the image in the center
+ */
+
+export function getCenteredCropParams(
+  imageDimensions: ImageDimensions,
+  cropDimensions: ImageDimensions,
+): { originX: number; originY: number; width: number; height: number } {
```
Comment (aussedatlo) [inline apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getCenteredCropParams.ts:14]: [COULD] create a type for `{ originX: number; originY: number; width: number; height: number }`

File: apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getCenteredCropParams.test.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { getCenteredCropParams } from "./getCenteredCropParams";
+
+describe("getCenteredCropParams", () => {
+  it("returns the correct crop params when the image is bigger than the container", () => {
+    expect(getCenteredCropParams({ width: 200, height: 100 }, { width: 100, height: 100 })).toEqual(
```
Comment (aussedatlo) [inline apps/ledger-live-mobile/src/components/CustomImage/useCenteredImage/getCenteredCropParams.test.ts:5]: [COULD] Add intermediate variables for better readability

for example:
```
const params = getCenteredCropParams({ width: 200, height: 100 }, { width: 100, height: 100 });
expect(params).toEqual(
      {
        width: 100,
        height: 100,
        originX: 50,
        originY: 0,
      },
    );
```

**Review by aussedatlo — APPROVED** [review]
LGTM 👍

**Review by aussedatlo — APPROVED** [review]

**Review by KVNLS — APPROVED** [review]

**Pull request merged**